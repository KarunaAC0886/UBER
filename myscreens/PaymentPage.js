import { View, Text, StatusBar, TouchableOpacity, Image } from 'react-native'
import React from 'react'
import BackNavigator from './BackNavigator'
import styles from './styles'
const PaymentPage = () => {
  return (
    <View style={styles.Paymentpg_main}>
      <BackNavigator titleName="Payment" navpath="account"/>
      <View style={styles.Paymentpg_frst_cont}>
        <View style={styles.Paymentpg_frst_adpaymnt}>
          <TouchableOpacity>
            <Text style={styles.Paymentpg_frst_adpaymnt_lbl}>Add Payment Method</Text>
          </TouchableOpacity>
        </View>
        <View style={styles.Paymentpg_scnd_cont}>
          <Text style={styles.Paymentpg_scndcont_frsttxt}>Ride Profiles</Text>
          <TouchableOpacity style={styles.Paymentpg_scnd_prflone}>
            <Image style={styles.Paymentpg_scnd_prflimg} source={require("../assets/Ellipse9.png")} />
            <View style={styles.Paymentpg_scnd_prfl_prsnl_cont}>
              <Text style={styles.Paymentpg_scnd_prfl_prsnl_txt}>Personal </Text>
            </View>
          </TouchableOpacity>
          <TouchableOpacity style={styles.Paymentpg_scnd_prflone}>
            <Image style={styles.Paymentpg_scnd_prflimg} source={require("../assets/Ellipse10.png")} />
            <View style={styles.Paymentpg_scnd_prfl_prsnl_cont}>
              <Text style={styles.Paymentpg_scnd_prfl_bsns_txt}>Start using Uber for business </Text>
              <Text style={styles.Paymentpg_scnd_prfl_bsns_txt2}>Turn on business travel features</Text>
            </View>
          </TouchableOpacity>
        </View>
        <View style={styles.Paymentpg_trd_cont}>
          <Text style={styles.Paymentpg_trd_frsttxt}>Promotions</Text>
          <TouchableOpacity style={styles.Paymentpg_trd_scnd_cont}>
            <Image style={styles.Paymentpg_trd_promoimg} source={require('../assets/Rectangle16.png')} />
            <Text style={styles.Paymentpg_trd_prom_secttl}>Promotions</Text>
          </TouchableOpacity>
          <TouchableOpacity style={styles.Paymentpg_trd_addpromo_btn}>
          <Text style={styles.Paymentpg_trd_addpromo_txt}>Add Promo code</Text>
          </TouchableOpacity>
        </View>
        <View style={styles.Paymentpg_fourt_cont}>
        
          <Text style={styles.Paymentpg_trd_frsttxt}>Vouchers</Text>
          <TouchableOpacity style={styles.Paymentpg_trd_scnd_cont}>
            <Image style={styles.Paymentpg_trd_promoimg} source={require('../assets/Voucher.png')} />
            <Text style={styles.Paymentpg_trd_prom_secttl}>Vouchers</Text>
          </TouchableOpacity>
          <TouchableOpacity style={styles.Paymentpg_trd_addpromo_btn}>
          <Text style={styles.Paymentpg_trd_addpromo_txt}>Add Voucher code</Text>
          </TouchableOpacity>
        
        </View>
      </View>
      <StatusBar />
    </View>
  )
}

export default PaymentPage;