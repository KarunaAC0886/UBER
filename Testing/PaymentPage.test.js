import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { useNavigation } from '@react-navigation/native';
import PaymentPage from '../myscreens/PaymentPage';

jest.mock('@expo/vector-icons', () => ({
    FontAwesome5: 'FontAwesome5',
}));

jest.mock('react-native-responsive-screen', () => ({
  widthPercentageToDP: jest.fn(),
  heightPercentageToDP: jest.fn(),
}));

jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));

describe('PaymentPage', () => {
  it('PaymentPage page renders correctly', () => {
    useNavigation.mockReturnValue({ navigate: jest.fn() });

    const tree = renderer.create(<PaymentPage/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});