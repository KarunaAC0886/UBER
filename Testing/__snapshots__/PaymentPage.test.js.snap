// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentPage PaymentPage page renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#000",
      "flex": 0,
      "flexDirection": "column",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#000000",
        "height": "13%",
        "justifyContent": "space-between",
        "paddingBottom": "5%",
        "paddingHorizontal": "6%",
        "paddingVertical": "2%",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <FontAwesome5
        name="arrow-left"
        style={
          {
            "color": "#ffffff",
            "fontSize": 20,
          }
        }
      />
    </View>
    <Text
      style={
        {
          "color": "#ffffff",
          "fontFamily": "Roboto",
          "fontSize": 36,
        }
      }
    >
      Payment
    </Text>
  </View>
  <View
    style={
      {
        "backgroundColor": "#1A1A1A",
        "flex": 1,
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "borderBottomColor": "#979797",
          "borderBottomWidth": 0.2,
          "height": "12%",
          "justifyContent": "center",
          "paddingHorizontal": "6%",
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "color": "#535AFF",
              "fontSize": 18,
              "fontWeight": "400",
            }
          }
        >
          Add Payment Method
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#979797",
          "borderBottomWidth": 0.2,
          "height": "34%",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#EDF6FF",
            "fontSize": 18,
            "fontWeight": "400",
            "paddingHorizontal": "6%",
            "paddingTop": "3%",
          }
        }
      >
        Ride Profiles
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flex": 0,
            "flexDirection": "row",
            "height": "40%",
            "opacity": 1,
            "paddingHorizontal": "6%",
            "width": "100%",
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/Ellipse9.png",
            }
          }
          style={
            {
              "flex": 0,
              "height": 60,
              "marginTop": "4%",
              "width": 60,
            }
          }
        />
        <View
          style={
            {
              "flex": 0,
              "flexDirection": "column",
              "justifyContent": "center",
              "paddingLeft": "10%",
              "width": "78%",
            }
          }
        >
          <Text
            style={
              {
                "color": "#EDF6FF",
                "fontSize": 18,
                "fontWeight": "400",
                "width": "100%",
              }
            }
          >
            Personal 
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flex": 0,
            "flexDirection": "row",
            "height": "40%",
            "opacity": 1,
            "paddingHorizontal": "6%",
            "width": "100%",
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/Ellipse10.png",
            }
          }
          style={
            {
              "flex": 0,
              "height": 60,
              "marginTop": "4%",
              "width": 60,
            }
          }
        />
        <View
          style={
            {
              "flex": 0,
              "flexDirection": "column",
              "justifyContent": "center",
              "paddingLeft": "10%",
              "width": "78%",
            }
          }
        >
          <Text
            style={
              {
                "color": "#535AFF",
                "fontSize": 18,
              }
            }
          >
            Start using Uber for business 
          </Text>
          <Text
            style={
              {
                "color": "#DADADA",
                "fontSize": 14,
                "fontWeight": "400",
              }
            }
          >
            Turn on business travel features
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#979797",
          "borderBottomWidth": 0.2,
          "height": "20%",
          "padding": "2%",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#EDF6FF",
            "fontSize": 18,
            "fontWeight": "400",
            "paddingLeft": "6%",
          }
        }
      >
        Promotions
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flex": 0,
            "flexDirection": "row",
            "height": "40%",
            "marginTop": "4%",
            "opacity": 1,
            "paddingLeft": "6%",
            "width": "100%",
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/Rectangle16.png",
            }
          }
          style={
            {
              "height": "75%",
              "width": "15%",
            }
          }
        />
        <Text
          style={
            {
              "color": "#EDF6FF",
              "fontSize": 18,
              "fontWeight": "400",
              "paddingLeft": "8%",
              "paddingTop": "2%",
            }
          }
        >
          Promotions
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
            "paddingLeft": "6%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#535AFF",
              "fontSize": 18,
              "fontWeight": "400",
            }
          }
        >
          Add Promo code
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "height": "20%",
          "padding": "2%",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#EDF6FF",
            "fontSize": 18,
            "fontWeight": "400",
            "paddingLeft": "6%",
          }
        }
      >
        Vouchers
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flex": 0,
            "flexDirection": "row",
            "height": "40%",
            "marginTop": "4%",
            "opacity": 1,
            "paddingLeft": "6%",
            "width": "100%",
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/Voucher.png",
            }
          }
          style={
            {
              "height": "75%",
              "width": "15%",
            }
          }
        />
        <Text
          style={
            {
              "color": "#EDF6FF",
              "fontSize": 18,
              "fontWeight": "400",
              "paddingLeft": "8%",
              "paddingTop": "2%",
            }
          }
        >
          Vouchers
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
            "paddingLeft": "6%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#535AFF",
              "fontSize": 18,
              "fontWeight": "400",
            }
          }
        >
          Add Voucher code
        </Text>
      </View>
    </View>
  </View>
</View>
`;
